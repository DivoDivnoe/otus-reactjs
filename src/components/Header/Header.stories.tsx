import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Header } from './Header';

export default {
  title: 'Example/Header',
  component: Header,
  argTypes: {
    signout: { action: 'signout' },
  },
} as ComponentMeta<typeof Header>;

const Template: ComponentStory<typeof Header> = (args) => <Header {...args} />;

export const Default = Template.bind(this);
Default.args = { user: null };

export const withUser = Template.bind(this);
Default.args = { user: 'Andrey' };
