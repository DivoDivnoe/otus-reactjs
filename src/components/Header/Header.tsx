import React, { FC } from 'react';
import styled from '@emotion/styled';
import { UserType } from '@/modules/user';

const Wrapper = styled.div`
  align-self: stretch;
`;

const Title = styled.h1`
  display: flex;
  justify-content: space-between;
  color: #ffffff;
`;

const LogoutItem = styled.span`
  cursor: pointer;

  &:hover {
    color: #058aba;
  }
`;

export interface HeaderPropsType {
  /**
   * user name
   */
  user: UserType;
  /**
   * dispatches logout action
   */
  signout: () => void;
}

export const Header: FC<HeaderPropsType> = ({ user, signout }) => {
  return (
    <Wrapper>
      {user && (
        <Title>
          <span data-testid='user'>Hello, {user}! </span>
          <LogoutItem onClick={signout}>Logout</LogoutItem>
        </Title>
      )}
    </Wrapper>
  );
};
