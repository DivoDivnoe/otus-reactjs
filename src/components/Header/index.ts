export { Header, HeaderPropsType } from './Header';
