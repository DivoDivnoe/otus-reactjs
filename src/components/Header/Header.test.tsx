import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { Header, HeaderPropsType } from './Header';

describe('Header', () => {
  it('renders correctly', () => {
    const mocks: HeaderPropsType = {
      user: 'Andrey',
      signout: jest.fn(),
    };
    const { asFragment } = render(<Header {...mocks} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('handles click event correctly', () => {
    const mocks: HeaderPropsType = {
      user: 'Andrey',
      signout: jest.fn(),
    };
    render(<Header {...mocks} />);

    fireEvent.click(screen.getByText('Logout'));
    expect(mocks.signout).toHaveBeenCalledTimes(1);
  });
});
