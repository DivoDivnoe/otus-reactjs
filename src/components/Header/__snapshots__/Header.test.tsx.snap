// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-quy5tq"
  >
    <h1
      class="css-hfi35m"
    >
      <span
        data-testid="user"
      >
        Hello, Andrey! 
      </span>
      <span
        class="css-1b0a5un"
      >
        Logout
      </span>
    </h1>
  </div>
</DocumentFragment>
`;
